src/resources/triton_python_backend_utils.py

```diff
@@ -497,6 +498,27 @@ def add_output(self, output):
 
         self._model_config["output"].append(output)
 
+logger = Logger()
+
+def log(message, level, nested_call=False):
+    logger.log(message, level, nested_call)
+
+
+def log_info(message):
```

Comment:
Move all this functions to C++.

Reply:
The python client now has access to a singleton from C++, but no ownership. The client can now call log functions from their models in the following fashion:
```
logger = pb_utils.Logger.instance()
logger.log("Hello!", logger.INFO)
logger.log_info("Info Msg!")
logger.log_warn("Warning Msg!")
logger.log_error("Error Msg!")
logger.log_verbose("Verbose Msg!")
```

Reply:
I think the API needs some improvements. How about not exposing `instance()` to the user? We set the pb_utils.Logger to the instance.

Then the API would look like:

```
logger = pb_utils.Logger
logger.log("Specific Msg!", logger.INFO)
logger.log_info("Info Msg!")
logger.log_warn("Warning Msg!")
logger.log_error("Error Msg!")
logger.log_verbose("Verbose Msg!")
```

Reply:
It's been tricky attempting to bind a property with the same name as the class. Is it appropriate to tie the instance to a non-function property of the class (e.g. `pb_utils.Logger.Log`)? Bound like so: `def_property_readonly_static("Log", [](py::object) { return &Logger::Instance(); });`